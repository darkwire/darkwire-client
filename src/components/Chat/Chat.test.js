import React from 'react'
import renderer from 'react-test-renderer'
import { Chat } from './index.js'

const sendSocketMessage = jest.fn()

test('Chat Component', () => {
  const component = renderer.create(
    <Chat focusChat={false} userId="foo" username="user" showNotice={() => {}} clearActivities={() => {}} sendSocketMessage={sendSocketMessage} />
  )
  const componentJSON = component.toJSON()

  expect(component).toMatchSnapshot()
  expect(componentJSON.children.length).toBe(2)
})
