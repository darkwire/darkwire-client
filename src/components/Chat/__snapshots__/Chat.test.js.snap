// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display table when data provided. 1`] = `
<form
  className="chat-preflight-container"
  onSubmit={[Function]}
>
  <input
    autoFocus="autofocus"
    className="chat"
    onChange={[Function]}
    placeholder="Type here"
    type="text"
    value=""
  />
  <div
    className="input-controls"
  >
    <button
      className="icon is-right send btn btn-link"
      onClick={[Function]}
    >
      <svg
        className="disabled"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polyline
          fill="none"
          points="15 10 20 15 15 20"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
        />
        <path
          d="M4 4v7a4 4 0 0 0 4 4h12"
          fill="none"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
        />
      </svg>
    </button>
  </div>
</form>
`;
